File: handlers/fasttopay.py
# use/bin/env python3
#-*-coding:utf-8-*-
import os
import tornado.web
import base64
import logging
from allsocrepydemo.methods.rsautill  import encryptbypublic
from allsocrepydemo.methods.urltool import urlencoding
from allsocrepydemo.methods.ordersign import ordersort,ordersign
from allsocrepydemo.methods.httpposttool import dopost
from allsocrepydemo.handlers.urladr import fastpay
from allsocrepydemo.methods.parasexml import fastOrderxml
class FastToPay(tornado.web.RequestHandler):
    def post(self):
        pathpublic = os.path.abspath("allscore_public_key.pem")
        pathprivate = os.path.abspath("rsa_private_key.pem")
        fastpaydict = {
            "merchantId": self.get_argument("merchantId"),
            "transAmt": self.get_argument("transAmt"),
            "outOrderId": self.get_argument("outOrderId"),
            "outAcctId": self.get_argument("outAcctId"),
            "payMethod": self.get_argument("payMethod"),
            "inputCharset": self.get_argument("inputCharset"),
            "signType": self.get_argument("signType"),
            "service": self.get_argument("service"),
            "orderId": self.get_argument("orderId"),
            "verifyCode": encryptbypublic(self.get_argument("verifyCode"), pathpublic).decode("utf-8"),
        }
        fasttopaystr=ordersort(urlencoding(fastpaydict))
        sign=ordersign(fastpaydict,pathprivate)
        logging.info(sign)
        sign=base64.b64encode(sign).decode("utf-8")
        fasttopaystr=fasttopaystr+"&sign="+sign
        logging.info(fasttopaystr)
        data=dopost(fasttopaystr,fastpay)
        logging.info(data)
        repdict=fastOrderxml(data)
        logging.info(repdict)

File: statics/js/fasttopay.js
$(document).ready(function(){
    $("#submit").click(function(){
        var merchantId = $("#merchantId").val();
        var inputCharset = $("#inputCharset").val();
        var service = $("#service").val();
        var transAmt = $("#transAmt").val();
        var outOrderId = $("#outOrderId").val();
        var outAcctId = $("#outAcctId").val();
        var payMethod = $("#payMethod").val();
        var orderId = $("#orderId").val();
        var signType = $("#signType").val();
        var verifyCode = $("#verifyCode").val();
        var order={
            "merchantId":merchantId,
            "transAmt":transAmt,
            "outOrderId":outOrderId,
            "outAcctId":outAcctId,
            "payMethod":payMethod,
            "orderId":orderId,
            "inputCharset":inputCharset,
            "signType":signType,
            "service":service,
            "verifyCode":verifyCode
        };
        $.ajax({
            type:"post",
            url:"/fastpay",
            data:order,
            cache:false,
            success:function(data){
                 $(document.body).html(data);
            },
            error:function(){
                alert("error!");
            },
        });
    });
});